### ISSUE TITLE
Stats not updating after document delete by filter.

### ISSUE THREAD

**Issue by tnamao** [meilisearch/meilisearch#4018]
Hi, I always use the product conveniently.

**Describe the bug**

I'm using meilisearch v1.3.2 and I found a bug.
When document is deleted by filter, stats is not changed.
`Delete all documents` or `Delete one document` is working fine.

**To Reproduce**
- Add one docunment

```sh
curl \
  -X PUT http://127.0.0.1:7700/indexes/samples/documents \
  -H "Authorization: Bearer masterKey" \
  -H "Content-Type: application/json" \
  --data-binary '{"id":"abc","num":1,"content":"this document remain stats"}'

curl \
  -X GET http://127.0.0.1:7700/indexes/samples/documents \
  -H "Authorization: Bearer masterKey"
```

- Return one document

```
{
  "results": [
    {
      "id": "abc",
      "num": 1,
      "content": "this document remain stats"
    }
  ],
  "offset": 0,
  "limit": 20,
  "total": 1
}
```

- Delete document by filter

```sh
curl \
  -X PATCH http://127.0.0.1:7700/indexes/samples/settings \
  -H "Authorization: Bearer masterKey" \
  -H 'Content-Type: application/json' \
  --data-binary '{
    "filterableAttributes": ["num"]
  }'

curl \
  -X POST http://127.0.0.1:7700/indexes/samples/documents/delete \
  -H "Authorization: Bearer masterKey" \
  -H 'Content-Type: application/json' \
  --data-binary '{"filter":"num = 1"}'
```

- Deleted document is not returned

```json
{
  "results": [],
  "offset": 0,
  "limit": 20,
  "total": 0
}
```

- But stats is not changed (bug?)

```sh
curl \
  -X GET http://127.0.0.1:7700/stats \
  -H "Authorization: Bearer masterKey" \
  -H 'Content-Type: application/json'
```

```json
...
  "samples": {
    "numberOfDocuments": 1,
    "isIndexing": false,
    "fieldDistribution": {
      "content": 1,
      "id": 1,
      "num": 1
    }
  }
...
```

**Expected behavior**
I hope the stats are updated.

**Meilisearch version:**
- Meilisearch v1.3.2 (getmeili/meilisearch:v1.3.2)
- running on docker

**Comment by Kerollmops** [comment]
Hey @tnamao 👋

Thank you very much for this detailed bug report. I updated your report as you were using a `DELETE` instead of a `POST` when sending the _delete by filter_ body, but the bug is indeed present in v1.3.2 and probably older releases.

**Comment by curquiza** [comment]
@meilisearch/engine-team this bug is now a priority for v1.4.0
We also need to know since when the bug exists (delete by filter feature has been introduced in v1.2.0), and let the Cloud team know

**Comment by curquiza** [comment]
Fixed by #4053 
Will be released in v1.4.0-rc.2 in the coming week and then in v1.4.0 on September 25th

Also, for our internal needs, we will release two other patch releases containing this fix
- v1.2.1
- v1.3.4
